src/components/dataTable/js/tableCellDirective.js

```diff
@@ -0,0 +1,213 @@
+angular
+  .module('material.components.table')
+  .directive('mdCell', MdTableCellDirective)
+  .directive('mdAlign', MdTableAlignDirective)
+  .directive('mdOrderBy', MdTableOrderByDirective);
+
+
+function MdTableCellDirective() {
+
+  return {
+    restrict: 'A',
+    require: ['mdCell', '^^mdTable'],
+    controller: MdTableCellController,
+    controllerAs: '$mdCell',
+    link: postLink
+  };
+
+  function postLink(scope, element, attrs, ctrls) {
+    var ctrl = ctrls[0];
+    var tableCtrl = ctrls[1];
+
+    ctrl.initCtrl(tableCtrl);
+  }
+}
+
+function MdTableCellController($element, $scope) {
```

Comment:
We can avoid those things, but it is way clean and clear to follow the prototype pattern.

Also allows us to easily generate definitions from it. And provides consistency and possibilities for API access.